lib/trino-hdfs/src/main/java/io/trino/hdfs/util/NodeProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hdfs.util;
+
+import io.trino.spi.HostAddress;
+
+import java.util.Optional;
+
+public interface NodeProvider
+{
+    Optional<HostAddress> getHost(String identifier);
```

Comment:
Could this method be changed to
```suggestion
    List<HostAddress> getHosts(String identifier);
```
to allow for more flexible caching strategies?

Reply:
This is resolved.